### PR TITLE
bpo-43024: improve signature (in help, etc) for functions taking sent…

### PR DESCRIPTION
**Pull request by iritkatriel**
…inel defaults

https://bugs.python.org/issue43024

Automerge-Triggered-By: GH:pitrou

### CODE REVIEW

File: Lib/traceback.py [hunk 1]
```diff
@@ -84,8 +84,11 @@ def extract_tb(tb, limit=None):
     "another exception occurred:\n\n")
 
 
-_sentinel = object()
+class _Sentinel:
+    def __repr__(self):
+        return "<optional>"
```
Comment (iritkatriel) [inline Lib/traceback.py]: Maybe `"<derived>" `(or something else?) instead of `"<optional>"` .

Comment (taleinat) [reply Lib/traceback.py]: How about "\<not specified\>" or "\<unspecified\>"?

**Comment by iritkatriel** [comment]
@pitrou - Looks like we settled on "<implicit>" as the best option.   If we have a better pattern following @taleinat 's work we can change it later, but for now this is an improvement.

**Review by pitrou — APPROVED** [review]
I agree with the choice @iritkatriel . Thank you!

**Pull request merged**